discussion/filter.go

```diff
@@ -0,0 +1,64 @@
+package discussion
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/go-playground/validator/v10"
+)
+
+var validate *validator.Validate
+
+func init() {
+	validate = validator.New()
+	validate.RegisterTagNameFunc(func(fld reflect.StructField) string {
+		name := strings.SplitN(fld.Tag.Get("json"), ",", 2)[0]
+		if name == "-" {
+			return ""
+		}
+		return name
+	})
+}
+
+type Filter struct {
+	Type          string `json:"type" validate:"omitempty,oneof=openended issues qanda all"`
+	State         string
```

Comment:
like `Type` should we also validate `State`?

Reply:
Oh yes thanks for pointing out, previously state would depend on the type, but now it is possible to check it since all type will have the same state names.

Reply:
Oh yes thanks for pointing out, previously state would depend on the type, but now it is possible to check it since all type will have the same state names.